src/cds_objects.cc

```diff
@@ -101,11 +101,8 @@ int CdsObject::resourcesEqual(const std::shared_ptr<CdsObject>& obj)
         return 0;
 
     // compare all resources
-    for (size_t i = 0; i < resources.size(); i++) {
-        if (!resources[i]->equals(obj->resources[i]))
-            return 0;
-    }
-    return 1;
+    return std::all_of(resources.begin(), resources.end(), [&](const auto& res) //
+        { return res == obj->resources[std::distance(resources.begin(), resources.end())]; });
```

Comment:
Also tip: `distance` is constant but you evaluate it in every iteration and it's O(n). Better do it just once in the beginning:
`[r_to_find = obj->resources [distance(...)]](auto & res) {return res->equals(r_to_find);}`
(If I don't miss anything...)

Also I noticed that you dropped `equals` and now looks like you just compare pointers themselves...